vcd/metadata.go

```diff
@@ -321,9 +368,9 @@ func checkIgnoredMetadataConflicts(d *schema.ResourceData, vcdClient *VCDClient,
 	return nil
 }
 
-// updateMetadataInState updates metadata and metadata_entry in the Terraform state for the given receiver object.
+// updateMetadataInStateDeprecated updates deprecated metadata and the new metadata_entry in the Terraform state for the given receiver object.
 // This can be done as both are Computed, for compatibility reasons.
```

Comment:
Maybe we should add a TODO comment reminder on how/why/when to remove this function as discussed here:
https://github.com/vmware/terraform-provider-vcd/pull/1126#discussion_r1367143095

Reply:
Done! Also added some notes to some untouched functions, I think it clarifies what to do in the future.

Reply:
Thank you!